### DISCUSSION TITLE
How to init display as fast as possible?

### DISCUSSION THREAD

**Discussion by stevempotter** [Bodmer/TFT_eSPI#2217] [Q&A - General]
I am using TFT_eSPI for the digital readout of a CNC pendant. This gets switched off and on a lot. I want the coordinates to be displayed as fast as possible. Presently it takes 567 ms (over half a second) for the device to initialize (just the TFT part). Is there any way I can speed that up so that the screen lights up with numbers within 200ms or less after power up?  There are use cases where this half-second delay causes an unsafe condition (needing to hit STOP as quickly as possible to halt the machine).
I am using a ST7789 display on the TTGO LilyGO T-Display V1, which is an ESP32 development board.

**Comment by Bodmer** [comment]
The display has to be reset and configured at power up. The ST7789 data sheet specifies the delays needed at each stage. You could experiment with the delays to see if shorter ones work but the reliability may suffer.

Delays are in the init code here:
https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_Drivers/ST7789_Init.h

and the reset delays here:

https://github.com/Bodmer/TFT_eSPI/blob/master/TFT_eSPI.cpp#L690
